test/ConfigProviderTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DotTest\Controller;
+
+use Dot\Controller\ConfigProvider;
+use Dot\Controller\Factory\ControllerEventListenersInitializer;
+use Dot\Controller\Factory\PluginManagerAwareInitializer;
+use Dot\Controller\Plugin\PluginManager;
+use PHPUnit\Framework\TestCase;
+
+class ConfigProviderTest extends TestCase
+{
+    protected array $config;
+
+    protected function setup(): void
+    {
+        $this->config = (new ConfigProvider())();
+    }
+
+    public function testHasDependencies(): void
+    {
+        $this->assertArrayHasKey('dependencies', $this->config);
+        $this->assertArrayHasKey('dot_controller', $this->config);
+        $this->assertArrayHasKey('plugin_manager', $this->config['dot_controller']);
+        $this->assertArrayHasKey('event_listeners', $this->config['dot_controller']);
+    }
+
+    public function testDependenciesHasFactories(): void
```

Comment:
Please break this into 2 tests:
* keep `testDependenciesHasFactories` for just the factory checks
* add `testDependenciesHasInitializers` for just the initializer checks